### ISSUE TITLE
Octal literals are not allowed in template strings. But how can I use \2 to match the 2 parenthetical in capturing groups of JavaScript's RegExp

### ISSUE THREAD

**Issue by LucaslEliane** [eslint/eslint#9128]
**Tell us about your environment**

* **ESLint Version: v3.19.0**
* **Node Version: v7.3.0**
* **npm Version: 3.10.10**

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

module.exports = {
    "extends": "airbnb-base",
    "plugins": [
        "import"
    ]
};

```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js


const attribute = new RegExp(`^\s*(${tname})[\s*=\s*((?:[\"\'])[^\"]*\")\3]?`);


```

**What did you expect to happen?**

when I use the SEEMING octal literals, actually I need use that to match the n parenthetical in capturing groups. LOL

**What actually happened? Please include the actual, raw output from ESLint.**

SyntaxError: Octal literals are not allowed in template strings.
    at Object.exports.runInThisContext (vm.js:78:16)
    at Module._compile (module.js:543:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:420:7)
    at startup (bootstrap_node.js:139:9)
    at bootstrap_node.js:535:3

**Comment by not-an-aardvark** [comment]
It looks like you want to pass a string containing a backslash to the `RegExp` constructor. To represent a string containing a backslash in a template literal, you should escape the backslash.

```js
const attribute = new RegExp(`^\s*(${tname})[\s*=\s*((?:[\"\'])[^\"]*\")\\3]?`);
```

This issue doesn't really have anything to do with ESLint (the stack trace you posted comes from Node trying to execute your file, not ESLint parsing it). I would recommend asking general JavaScript questions on a site like StackOverflow.

**Comment by LucaslEliane** [comment]
Oh! Sorry, I made a mistake.
Thanks for your reply.
QAQ
